### PR TITLE
Enable PGF to compile.

### PR DESCRIPTION
**Pull request by DJDavies2**
## Description

Currently GSW doesn't compile with pgfortran, failing like this:

PGF90-S-0155-DATA initialization with nonconstant expression  (/home/h01/frwd/cylc-run/mi-be984/work/1/git_clone_gsw_fortran/GSW-Fortran/modules/gsw_mod_check_data.F90: 784)
.....

This change adds some #ifdefs that enable this to compile with pgfortran.  It isn't very pretty, if there is a better way to arrange this I can do so.

## Issue(s) addressed

None, I didn't see how to open an issue?

## Dependencies

None.

## Impact

None.

## Checklist

- [x ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x ] I have run the unit tests before creating the PR

### CODE REVIEW

**Comment by climbfuji** [comment]
@eap Can you please take a look at the CI failures here (I realize we didn't haveany PRs for gsw-fortran since the new CI system was put in place)? Something isn't quite right:
```
+ /workdir/CI/src/configure_bundle/configure_bundle.py --unit-test --test-target GSW-Fortran --unit-test-dependency --dependency-version crtm=788756fb773de0ef5da255ee0f84f81559daeb89 coupling=c13535676ecd95de81492c2a534a70064c41868a femps=cb396811eb26380478c4d3f177d95096ed2ddd8f fv3-jedi=6f5fe3f87ed4ca8202bc89e11c6c37ad73154b84 fv3-jedi-data=5db99d77f59c7d55b085e2650b19e53578cc1d7f fv3-jedi-lm=47462e74705f7de7ad13559ec749acd26d987e3f fv3=61450b4e3e80bb96b26c5f3808ce60b5e5cb4207 gsw=489a788fcc4b241c9713fd3b321d457379c571aa ioda=5e54d603a437bfe4138951bafe76a4a1557b9db3 iodaconv=a2f420d73d7cd0e69dfff833eefc74e71e71ee2d ioda-data=60d8955743b0f9e15e8b148deccfe7a3d1929653 mom6=51ec489ad7d8a86762bef4c46eabd9af5fc41fa4 mpas=3ecd59e2cdeadf5e3a1d6a11aa14a4905e7f60a2 mpas-jedi=1cb82ea2847d3e30adf5384d0add6f9cbcbcc1f8 mpas-jedi-data=bbe282ddc90d6e72fa9843dc5ca2f47b9227bd2a oops=2012398058043bda46db2b9d80a909810df04cea saber=ace19a0ee9789f507e10ca6951a5099981a8f706 soca=729e9b73c635f450fc817c3b2d1de24d3581f09a ufo=6944805c0912a81993adac7667031668f9056a5c ufo-data=62f4d464d7a9c50004746c85462de19abfbe73b7 vader=b8404a83208d46bbfb5add30b6561c54da9c6d82 --bundle-root=/workdir/test_root/jedi-bundle
Traceback (most recent call last):
  File "/workdir/CI/src/configure_bundle/configure_bundle.py", line 382, in <module>
    main(bundle_root=args.bundle_root,
  File "/workdir/CI/src/configure_bundle/configure_bundle.py", line 337, in main
    raise ValueError(
ValueError: Flag "--test-target GSW-Fortran" is invalid; package "GSW-Fortran"" is not used by this bundle.
+ '[' 1 -ne 0 ']'
+ util.check_run_fail JCSDA-internal/GSW-Fortran 22192760524 'Bundle configuration failed'
+ '[' no = yes ']'
+ repo=JCSDA-internal/GSW-Fortran
+ run_id=22192760524
+ fail_reason='Bundle configuration failed'
```

**Review by climbfuji — COMMENTED** [review]
ifdefs are never pretty, but that's just reality when you have to use a crappy compiler.

Personally, I would have preferred to flip the logic:
```
#ifdef PGF
...
#else
...
#endif
```
but it's not a big deal. Even better, those data files should be put in netcdf files rather than hardcoded in modules ... but that's beyond our control.

I'll wait with approving until the CI tests work. We should consider pushing this bug fix upstream, eventually we want to try using the authoritative repo instead of our fork.

**Comment by travissluka** [comment]
I'll take a closer look at this Monday, but ultimately we want to stop using gsw-fortran. UFO and Vader have C wrappers around the gsw-fortran  code, so we should just switch to gsw-c and have that as part of spack stack. We can still do this PR in the meantime as a bandaid

**Comment by eap** [comment]
@DJDavies2 - we might have a fix for this issue, can you push another commit (even an empty commit) to re-trigger the tests?

**Comment by DJDavies2** [comment]
I've pushed a commit.

**Comment by climbfuji** [comment]
> I didn't think I could create a branch in JCSDA-internal for this repository, perhaps I was mistaken? Should I try and see if that would help?

If you are ok to wait, then it would be better to fix the CI system so that it can handle forks.

**Comment by DJDavies2** [comment]
Okay, that is fine, there is no rush for this.

**Review by climbfuji — APPROVED** [review]

**Review by climbfuji — COMMENTED** [review]
@DJDavies2 I am running a final round of jedi-bundle CI tests with this. We'll merge it tomorrow unless there are any surprises and solve the branch-on-fork problem later (@eap will file a bug in the CI repo for this).

**Comment by climbfuji** [comment]
@travissluka Can you take a look at this PR, please? I think jedi-bundle ctest all look ok. The system I am running on (ParallelWorks Gcloud) gives different answers for some of the soca tests, which is probably unrelated - but better to check.

**Comment by travissluka** [comment]
> @travissluka Can you take a look at this PR, please? I think jedi-bundle ctest all look ok. The system I am running on (ParallelWorks Gcloud) gives different answers for some of the soca tests, which is probably unrelated - but better to check.

Will do, I'll check this morning

File: modules/gsw_mod_check_data.F90 [hunk 1]
```diff
@@ -42,7 +42,7 @@ module gsw_mod_check_data
     real (r8), dimension(cast_n) :: values
 end type gsw_result_cast
 
-! Different versions work for different compilers ... sigh.
+! Different versions work for different compilers .... sigh.
```
Comment (travissluka) [inline modules/gsw_mod_check_data.F90:45]: nice touch with the extra dot before the sigh :laughing:

**Review by travissluka — APPROVED** [review]

**Comment by climbfuji** [comment]
Thanks for your patience @DJDavies2 and thanks for testing @travissluka

File: modules/gsw_mod_check_data.F90 (hunk 1 above)
Comment (DJDavies2) [reply modules/gsw_mod_check_data.F90:45]: I just did that to trigger the CI.

**Pull request merged**